app.rb

```diff
@@ -22,25 +22,65 @@ def list_all_people(select: false)
     end
   end
 
-  def create_student(age, name, parent_permission)
-    @people << Student.new(age, name, parent_permission: parent_permission)
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.downcase == 'y'
+    @people << Student.new(age, name, parent_permission: permission)
+    puts 'Person created successfully'
   end
 
-  def create_teacher(age, name, specialization)
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
```

Comment:
Similar to the above comment and for the same reasons, it would be better that all requests for input and input validation be placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.

This is repeated in most of the other methods, kindly adjust them too.